cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_splitter_sp_opt_impl_dpc.cpp

```diff
@@ -226,11 +249,19 @@ sycl::event train_best_split_sp_opt_impl<Float, Bin, Index, Task, sbg_size>::
                         Index response_int = (i < row_count) ? static_cast<Index>(response) : -1;
 
                         ts.ftr_bin = -1;
-
+                        if (ctx.splitter_mode_value == splitter_mode::best) {
+                            ts.ftr_bin = ft_rnd_ptr[ftr_idx] % index_max;
+                        }
                         ts.ftr_bin = sycl::reduce_over_group(item.get_group(),
                                                              bin > ts.ftr_bin ? bin : index_max,
                                                              minimum<Index>());
```

Comment:
This logic checks that the chosen bin's index is not greater than the maximum bin's index in the workgroup. But in case of extra trees the bins of the histogram do not need to be processed in parallel, so this logic won't be needed in case of extra trees.

[But it might be a performance benefit from the processing the features in parallel, i.e. for at the line #237]